assets/js/googlesitekit/api/cache.test.js

```diff
@@ -0,0 +1,321 @@
+/**
+ * Internal dependencies
+ */
+// eslint-disable-next-line @wordpress/dependency-group
+import { _setStorageKeyPrefix, _setSelectedStorageBackend, get, set, deleteItem, getKeys, clearCache } from './cache';
+
+describe( 'googlesitekit.api.cache', () => {
+	[ 'localStorage', 'sessionStorage' ].forEach( ( backend ) => {
+		describe( `${ backend } backend`, () => {
+			let storageMechanism;
+			beforeAll( () => {
+				storageMechanism = global[ backend ];
+				_setSelectedStorageBackend( storageMechanism );
+			} );
+
+			afterAll( () => {
+				// Reset the backend storage mechanism.
+				_setSelectedStorageBackend( undefined );
+			} );
+
+			describe( 'get', () => {
+				it( 'should return undefined when the key is not found', async () => {
+					const result = await get( 'not-a-key' );
+
+					expect( storageMechanism.getItem ).toHaveBeenCalledWith( 'not-a-key' );
+					expect( result.cacheHit ).toEqual( false );
+					expect( result.value ).toEqual( undefined );
+				} );
+
+				it( 'should return undefined when the key is found but the TTL is too high', async () => {
+					const didSave = await set( 'old-key', 'something', 1 );
+					expect( didSave ).toEqual( true );
+
+					const result = await get( 'old-key', 5 );
+
+					expect( storageMechanism.getItem ).toHaveBeenCalledWith( 'old-key' );
+					expect( result.cacheHit ).toEqual( false );
+					expect( result.value ).toEqual( undefined );
+				} );
```

Comment:
This test is a bit misleading (although I'm sure it passes). Cache would never be a miss if the TTL is too high. A high TTL would be a cached value which is valid for a longer time, not less.

This is a bit confusing by the way the test is written because the cached timestamp is `1` (as in Jan 1, 1970 12:00:01 am), and the `old-key` is fetched with a max age of 5 seconds.

Since TTL is always relative to the current time (via `Date.now`) it would be more clear to use a timestamp relative to the current time, with an offset to show that it is clearly _older_ than the max age passed in `get`.

Can we tweak this to be a little more clear? 🙂